orchagent/orch.cpp

```diff
@@ -685,6 +685,46 @@ Executor *Orch::getExecutor(string executorName)
     return NULL;
 }
 
+bool Orch::handleSaiCreateFailure(sai_status_t status)
+{
+    /*
+     * This function aims to provide coarse handling of failures in sairedis create
+     * operation (i.e., notify users by throwing excepions when failures happen).
+     * TODO: 1. Add general handling logic for specific statuses (e.g., SAI_STATUS_ITEM_ALREADY_EXISTS)
+     *       2. Develop fine-grain failure handling mechanisms specific and replace
+     *          this coarse handling in each orch.
+     */
+    SWSS_LOG_THROW("Encountered failure in Create operation, status: %d", status);
+    return false;
```

Comment:
>false [](start = 11, length = 5)

above line with throw anyway, so this line is dead code? can we remove it?

Reply:
To make this function more general, only throw if `status != SAI_STATUS_SUCCESS`, and return true/false

---
In reply to: [560590840](https://github.com/Azure/sonic-swss/pull/1596#discussion_r560590840) [](ancestors = 560590840)

Reply:
And rename function `handleSaiCreateStatus`

---
In reply to: [560591437](https://github.com/Azure/sonic-swss/pull/1596#discussion_r560591437) [](ancestors = 560591437,560590840)

Reply:
Renamed the function to `handleSaiCreateStatus`. Also added sai api type as an input. For the return boolean value, I was thinking to use it as an indicator for whether the status is handled or not. Currently, our aim is to mimic the behavior as async mode, so no status is handled in the function and the return value is not very meaningful. As the function evolves, the situation should change.

Reply:
Could you clarify "use it as an indicator for whether the status is handled or not"? Do you assume `status != SAI_STATUS_SUCCESS` ? What is 'handled' ?
It should be good function level comment.


---
In reply to: [563944011](https://github.com/Azure/sonic-swss/pull/1596#discussion_r563944011) [](ancestors = 563944011)

Reply:
I think the criteria could be whether the SAI call needs to be retried after running this function. For example, similar to https://github.com/Azure/sonic-swss/blob/master/orchagent/neighorch.cpp#L639, it returns true if the create operation fails with the entry already exists. On the other hand, a false return value could mean retry is needed. Added a comment message to clarify. Also as it is assumed `status != SAI_STATUS_SUCCESS`, I added a check for `SAI_STATUS_SUCCESS` and log a warning when the function receives such a status.